lib/compress/zstd_compress.c

```diff
@@ -1112,19 +1142,30 @@ static void ZSTD_ldm_reduceTable(ldmEntry_t* const table, U32 const size,
 *   rescale all indexes to avoid future overflow (indexes are U32) */
 static void ZSTD_reduceIndex (ZSTD_CCtx* zc, const U32 reducerValue)
 {
-    { U32 const hSize = (U32)1 << zc->appliedParams.cParams.hashLog;
-      ZSTD_reduceTable(zc->hashTable, hSize, reducerValue); }
+    {   U32 const hSize = (U32)1 << zc->appliedParams.cParams.hashLog;
+        ZSTD_reduceTable(zc->hashTable, hSize, reducerValue);
+    }
+
+    if (zc->appliedParams.cParams.strategy != ZSTD_btlazy2) {
```

Comment:
Should this be deleted? It looks like for strategy that isn't `ZSTD_btlazy2` or `ZSTD_fast`, the chain table will be reduced twice.

Reply:
You are right, this is messed up.
As a mitigation, I suspect that the second `reduce` is a no-op, since it will discover it can't reduce indexes much further.
Still, it's useless cpu cycles spent.